src/app/clusters/occupancy-sensor-server/README.md

```diff
@@ -0,0 +1,170 @@
+Using `RegisteredServerCluster` simplifies registration.
+
+The `HoldTime` feature and its related attributes are optional. To enable them,
+call the `.WithHoldTime()` method on the `Config` object. If it is not called,
+these attributes will be disabled.
+
+```cpp
+#include "app/DefaultTimerDelegate.h"
+#include "app/clusters/occupancy-sensor-server/OccupancySensingCluster.h"
+#include "app/server-cluster/ServerClusterInterfaceRegistry.h"
+
+// In a .cpp file
+MyOccupancyDelegate gMyOccupancyDelegate;
+DefaultTimerDelegate gTimerDelegate;
+
+constexpr chip::app::Clusters::OccupancySensing::Structs::HoldTimeLimitsStruct::Type kHoldTimeLimits = {
+    .holdTimeMin = 1, .holdTimeMax = 300, .holdTimeDefault = 60
+};
+
+chip::app::RegisteredServerCluster<chip::app::Clusters::OccupancySensingCluster> gOccupancyCluster(
+    chip::app::Clusters::OccupancySensingCluster::Config(kYourEndpointId)
+        .WithFeatures(chip::app::Clusters::OccupancySensing::Feature::kPassiveInfrared)
+        .WithHoldTime(60, kHoldTimeLimits, gTimerDelegate) // Enable HoldTime feature
+        .WithDelegate(gMyOccupancyDelegate)                // Attach optional delegate
+);
+```
+
+### 3. Register the Cluster
+
+In your application's initialization sequence, register the cluster instance
+with either the `CodegenDataModelProvider` (legacy) or the
+`CodeDrivenDataModelProvider`.
+
+```cpp
+#include "data-model-providers/codegen/CodegenDataModelProvider.h"
+
+void ApplicationInit()
+{
+    // ... other initializations
+    CHIP_ERROR err = chip::app::CodegenDataModelProvider::Instance().Registry().Register(gOccupancyCluster.Registration());
+    // ...
+}
+```
+
+### 4. Notify the Cluster of Occupancy Changes
+
+Your application is responsible for monitoring the physical sensor. When the
+sensor state changes, call the `SetOccupancy()` method on the cluster instance.
+This must be called within the Matter context. Use `ScheduleWork` if calling
+from a separate application thread.
+
+```cpp
+void MySensorHardwareCallback(bool isOccupied)
+{
+    // Get the cluster instance (e.g., via the RegisteredServerCluster object)
+    chip::app::Clusters::OccupancySensingCluster * cluster = gOccupancyCluster.Get();
+    if (cluster)
+    {
+        cluster->SetOccupancy(isOccupied);
+    }
+}
+```
+
+## Legacy Ember-Style Usage
+
+For backwards compatibility with applications that rely on older ZAP-generated
+patterns (like the `all-clusters-app`), a compatibility layer is provided in
+`CodegenIntegration.h` and `CodegenIntegration.cpp`.
+
+In this model, you configure the Occupancy Sensing cluster in your `.zap` file.
+The ZAP tool generates `MatterOccupancySensingClusterInitCallback`, which is
+implemented by our `CodegenIntegration` layer to automatically instantiate and
+configure the cluster based on your ZAP configuration.
+
+To use the cluster in this mode, your application can get a pointer to the
```

Comment:
Should we mention here that moving forward we are aiming to deprecate this method of using clusters?